numpyro/infer/autoguide.py

```diff
@@ -1527,17 +1571,32 @@ def base_z_dist_log_prob(z):
                     base_z_dist_log_prob(z_0) / subsample_size,
                 )
             else:
-                z_0_loc_init = jnp.zeros((N, D))
-                z_0_loc = numpyro.param(
-                    "{}_z_0_loc".format(self.prefix), z_0_loc_init, event_dim=1
-                )
-                z_0_scale_init = jnp.ones((N, D)) * self.init_scale
-                z_0_scale = numpyro.param(
-                    "{}_z_0_scale".format(self.prefix),
-                    z_0_scale_init,
-                    constraint=constraints.positive,
-                    event_dim=1,
-                )
+                if self.use_global_dais_params:
+                    z_0_loc_init = jnp.zeros(D)
```

Comment:
i don't think this part makes sense. the `z` params should always be local.

Reply:
hmm, I think amortized guide should have `local_guide` specified.

Reply:
but what if i just want non-amortized mean-field variational distributions for the locals? i would need to specify `local_guide` as opposed to relying on a convenient default behavior?

Reply:
Yes, in that case, you might want to set `use_global_dais_params=False`. We might also change the semantics of this flag to `global_dais_params=None/"dynamic"/"full"`. wdyt?

Reply:
what exactly are the three options?

Reply:
None is the current master behavior, or False in this PR. dynamic is your request. full is this PR.

Reply:
sorry for being slow here (traveling). i think `use_global_dais_params` which controls e.g. `betas` should be entirely separate from whatever controls the distribution over `z_0`. what about the following behavior:
- if `local_guide` is provided use that.
- otherwise if `local_guide=None` and there exist local variables instantiate a auto mean-field guide?

Reply:
Hmm, that will imply that if users want to have global params for the base dist, they will need to use local_guide? I don't have a preference here. To me, base params play a similar role as betas; the dynamic will depend on model density etc. In the DAIS paper, the author even uses a fixed base dist.

Reply:
i don't think there's any reason in practice why you'd really want global/shared params for the base dist q(z_0). having global params for e.g. beta makes sense because it's basically a "higher order quantity" and so harder to estimate and probably varies less from data point to data point. just in the same way that we'd probably generally be more comfortable in sharing scales/variances across data points than locs/means.

Reply:
makes sense to me, thanks!